### PR TITLE
Data Science with Python : Complete Overview of Regression

### PR DESCRIPTION
**Pull request by yashh18**
<hr>

## Description 📜

Documentation contribution for Overview of Regression
Fixes : #735 

<hr>

## Type of change 📝

- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] Datascience with Python

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Comment by yashh18** [comment]
@kareem-negm @Yash621 @prathimacode-hub Please review my PR!

**Review by Yash621 — COMMENTED** [review]
Firstly make the readme file title bold ,after that make a separate folder for regression analysis under the folder of algorithms ,and place your regression overview file there.
secondly in your explanation you have only talked about linear regression and I can't find enough explanation on polynomial regression ,if you are mentioning a algorithm ,try to give a code example also for them.
and lastly try to include a brief overview of other types of regression like logistics ,lasso,ridge etc under the heading "some other imp types of regression algorithms".No need to give a code example for them just definition and formula.

**Comment by yashh18** [comment]
@Yash621 okay cool
So do i have to delete the whole folder and re-upload it or what's the process of editing it?

**Comment by prathimacode-hub** [comment]
That's upto you. You can do it afresh or do it it in your existing PR. @yashh18

**Review by prathimacode-hub — CHANGES_REQUESTED** [review]
Changes to be done 👍 

There is no sub repo of Documentation in Machine Learning. It should go in "Tutorials" folder
All images should be in "Images" folder
The project folder should be titled as "Overview of Regression"
The README.md file should be elaborate, it's not sufficient. It only has outline of it. You can add features, why its used, application, advantages, conclusion.
The readme file should be titled as "README.md"
In .ipynb file, you can add few more code snippets to represent the topics given. Also at end, overall summary of the topic.

Kindly work on it. @yashh18

File: Datascience_With_Python/Machine Learning/Overview_of_Regression/Documentation/overview_of_regression.ipynb [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+      "source": [
+        "#REGRESSION\r\n",
+        "\r\n",
+        "**Regression analysis** is a fundamental concept in the field of machine learning. It falls under supervised learning wherein the algorithm is trained with both input features and output labels. It helps in establishing a relationship among the variables by estimating how one variable affects the other.\r\n",
+        "\r\n",
+        "\r\n",
+        "Example: Suppose there is a marketing company A, who does various advertisement every year and get sales on that. The below list shows the advertisement made by the company in the last 5 years and the corresponding sales:\r\n",
+        "\r\n",
+        "\r\n",
+        "![image](images\\regression-analysis-in-machine-learning.png)\r\n",
```
Comment (Kareem-negm) [inline Datascience_With_Python/Machine Learning/Overview_of_Regression/Documentation/overview_of_regression.ipynb:41]: please fix this image URL too

File: Datascience_With_Python/Machine Learning/Overview_of_Regression/Documentation/overview_of_regression.ipynb [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+        "\r\n",
+        "$x_i$ is the input feature for $i^{th}$ value, \r\n",
+        "\r\n",
+        "$\\theta_i$ is the model parameter ($\\theta_0$ is the bias and the coefficients are $\\theta_1, \\theta_2, … \\theta_n$).\r\n",
+        "\r\n",
+        "The coefficient is like a volume knob, it varies according to the corresponding input attribute, which brings change in the final value. It signifies the contribution of the input variables in determining the best-fit line. \r\n",
+        "\r\n",
+        "Bias is a deviation induced to the line equation $y = mx$ for the predictions we make. We need to tune the bias to vary the position of the line that can fit best for the given data. \r\n",
+        "\r\n",
+        "![linreg](images\\linear.jpeg)\r\n",
```
Comment (Kareem-negm) [inline Datascience_With_Python/Machine Learning/Overview_of_Regression/Documentation/overview_of_regression.ipynb:110]: this too

File: Datascience_With_Python/Machine Learning/Overview_of_Regression/Documentation/overview_of_regression.ipynb [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+        "- Suppose there is a dataset which consists of datapoints which are present in a non-linear fashion, so for such case, linear regression will not best fit to those datapoints. To cover such datapoints, we need Polynomial regression.\r\n",
+        "- In Polynomial regression, the original features are transformed into polynomial features of given degree and then modeled using a linear model. Which means the datapoints are best fitted using a polynomial line.\r\n",
+        "\r\n",
+        "$$Y_{0} = b_{0}+ b_{1}x^{1} + … b_{n}x^{n}$$\r\n",
+        "\r\n",
+        "where $Y_{0}$ is the predicted value for the polynomial model with regression coefficients $b_{1}$ to $b_{n}$ for each degree and a bias of $b_{0}$.\r\n",
+        "\r\n",
+        "If n=1, the polynomial equation is said to be a linear equation.\r\n",
+        "\r\n",
+        "![pol](images\\pr7.png)"
```
Comment (Kareem-negm) [inline Datascience_With_Python/Machine Learning/Overview_of_Regression/Documentation/overview_of_regression.ipynb:290]: this too

**Comment by yashh18** [comment]
@prathimacode-hub Please review it once again !

**Review by prathimacode-hub — CHANGES_REQUESTED** [review]
Changes to be done :

Your README needs more detailing. This is a vast topic atleast a fair introduction related to this topic is needed. 
You don't have to add all your files in Documentation folder. Instead it can be uploaded in "Overview of Regression" as we have separate folders of Audio and Video
The project file name should be in snake case : overview_of_regression.ipynb
All the images should be in "Images" folder.

Kindly work on it. @yashh18

**Comment by yashh18** [comment]
@prathimacode-hub I have deleted the documentation folder and added the files in Overview of regression, changed the .ipynb file to snakecase, added all the images in the required folder and covered most of the information in both readme and jupyter file. 
So please review it once again !

**Review by prathimacode-hub — APPROVED** [review]
PR approved. 👍@yashh18

**Comment by yashh18** [comment]
@prathimacode-hub thank you ✨

**Pull request merged**